### PR TITLE
#1137 - Added tests for HostSelector and ScriptManager.

### PR DESCRIPTION
**Pull request by dudu84**
| Q             | A
| ------------- | ---
| Bug fix?      |  No
| New feature?  | No
| BC breaks?    |  No
| Deprecations? | No
| Fixed tickets | #1137 

> Do not forget to add notes about your changes to [CHANGELOG.md](https://github.com/deployphp/deployer/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by antonmedv** [comment]
Please, this code style. 
https://styleci.io/analyses/q5KNVx

File: test/src/Host/HostSelectorTest.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
+<?php
```
Comment (antonmedv) [inline test/src/Host/HostSelectorTest.php]: Add header comment (see other files)

File: test/src/Host/HostSelectorTest.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        $host = new  Host('server');
+        $host->roles( "role1");
+        $host->roles( "role2");
+        $hostCollection = new HostCollection();
+        $hostCollection->set('server', $host);
+        $hostSelector = new HostSelector($hostCollection);
+
+        $this->assertNotEmpty($hostSelector->getByRoles($roles));
+    }
+}
```
Comment (antonmedv) [inline test/src/Host/HostSelectorTest.php]: new line here

File: test/src/Task/ScriptManagerTest.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            ->onRoles('db');
+
+        $taskCollection = new TaskCollection();
+        $taskCollection->set('dump', $task);
+
+        $scriptManager = new ScriptManager($taskCollection, $hostCollection);
+
+        $this->assertNotEmpty($scriptManager->getTasks("dump"));
+    }
+}
```
Comment (antonmedv) [inline test/src/Task/ScriptManagerTest.php]: New line here.

**Comment by antonmedv** [comment]
Also please add test for case if host does not have any stage, skip them. See https://github.com/deployphp/deployer/blob/master/src/Host/HostSelector.php#L48

**Comment by antonmedv** [comment]
Also test for "if still is empty, try to find host by name", see https://github.com/deployphp/deployer/blob/master/src/Host/HostSelector.php#L56

**Comment by antonmedv** [comment]
All looks very well, nice work!

**Comment by dudu84** [comment]
Thanks!

**Comment by dudu84** [comment]
Hi! I've finished the requested changes.

**Comment by antonmedv** [comment]
Nice work! Congratulation with first contribution. 🙂

**Comment by dudu84** [comment]
Thanks, Anton! =)
I will contribute more for shure!

2017-05-26 13:08 GMT+02:00 Anton Medvedev <notifications@github.com>:

**Comment by reneroboter** [comment]
How do these tests work? The following test will fail because there is no config object available for the Deployer instance. I mean testReturnEmptyArrayOfHostsUsingGetByRolesIfNoRolesDefined. I found these issues because I currently working on #1610.
